qiime2/core/cache.py

```diff
@@ -465,15 +708,50 @@ def _register_key(self, key, value, pool=False):
             key_fp.write_text(_KEY_TEMPLATE % (key, value, ''))
 
     def load(self, key):
-        """Load the data pointed to by a key. Only works on a key that refers
-        to a data item will error on a key that points to a pool
+        """Loads the data pointed to by a key. Only works on keys that refer to
+        data items and will error on keys that refer to pools.
+
+        Parameters
+        ----------
+        key : str
+            The key to the data we are loading.
+
+        Returns
+        -------
+        Result
+            The loaded data pointed to by the key.
+
+        Raises
+        ------
+        ValueError
+            If the key does not reference any data meaning you probably tried
+            to load a pool.
+        ValueError
```

Comment:
@Oddant1, wrong error-mode, not-found should translate to `KeyError` whereas pool vs data makes more sense as a `ValueError` (since the key exists...)

Reply:
I was kinda thinking that, but I interpreted your initial comment as asking for the opposite and just didn't think about it too hard.

Reply:
No worries!